Tasks/DotNetCoreCLI/packcommand.ts

```diff
@@ -0,0 +1,162 @@
+                tl.debug(`Getting version from env var: ${versionEnvVar}`);
+                version = tl.getVariable(versionEnvVar);
+                if(!version)
+                {
+                    tl.setResult(tl.TaskResult.Failed, tl.loc("Error_NoValueFoundForEnvVar"));
+                    break;
+                }
+                break;
+            case "byBuildNumber":
+                tl.debug("Getting version number from build number")
+
+                if(tl.getVariable("SYSTEM_HOSTTYPE") === "release")
+                {
+                    tl.setResult(tl.TaskResult.Failed, tl.loc("Error_AutomaticallyVersionReleases"));
+                    return;
+                }
+
+                let buildNumber: string =  tl.getVariable("BUILD_BUILDNUMBER");
+                tl.debug(`Build number: ${buildNumber}`);
+
+                let versionRegex = /\d+\.\d+\.\d+(?:\.\d+)?/;
+                let versionMatches = buildNumber.match(versionRegex);
+                if (!versionMatches)
+                {
+                    tl.setResult(tl.TaskResult.Failed, tl.loc("Error_NoVersionFoundInBuildNumber"));
+                    return;
+                }
+
+                if (versionMatches.length > 1)
+                {
+                    tl.warning(tl.loc("Warning_MoreThanOneVersionInBuildNumber"))
+                }
+                
+                version = versionMatches[0];
+                break;
+        }
+
+        tl.debug(`Version to use: ${version}`);
+
+        if(outputDir && !tl.exist(outputDir))
+        {
+            tl.debug(`Creating output directory: ${outputDir}`);
+            tl.mkdirP(outputDir);
+        }
+
+        let useLegacyFind: boolean = tl.getVariable("NuGet.UseLegacyFindFiles") === "true";
+        let filesList: string[] = [];
+        if (!useLegacyFind) {
+            let findOptions: tl.FindOptions = <tl.FindOptions>{};
+            let matchOptions: tl.MatchOptions = <tl.MatchOptions>{};
+            filesList = tl.findMatch(undefined, searchPattern, findOptions, matchOptions);
+        }
+        else {
+            filesList = nutil.resolveFilterSpec(searchPattern);
+        }
+
+        tl.debug(`Found ${filesList.length} files`);
+        filesList.forEach(file => {
+            tl.debug(`--File: ${file}`);
+        });
+
+        let props: string[] = [];
+        if(configuration && configuration !== "$(BuildConfiguration)")
+        {
+            props.push(`Configuration=${configuration}`);
+        }
+        if(propertiesInput)
+        {
+            props = props.concat(propertiesInput.split(";"));
+        }
+
+        let environmentSettings: ngToolRunner.NuGetEnvironmentSettings = {
+            credProviderFolder: null,
+            extensionsDisabled: true
+        };
+
+        const dotnetPath = tl.which("dotnet", true);
+        for (const file of filesList) {
+            await dotnetPackAsync(dotnetPath, file, outputDir, nobuild, version, props, verbosity);
```

Comment:
good idea, but I'm worried interleaved output on async tool.exec would be confusing on error,